### PR TITLE
[WIP] OCPBUGS-30973: upgrade @types/react-redux

### PR DESCRIPTION
**Pull request by rhamilto**

### CODE REVIEW

File: frontend/packages/console-app/src/components/nodes/NodesPage.tsx [hunk 1]
```diff
@@ -241,7 +238,7 @@ const NodesTableRow: React.FC<RowProps<NodeKind, GetNodeStatusExtensions>> = ({
   rowData,
 }) => {
   const { t } = useTranslation();
-  const metrics = useSelector(({ UI }) => UI.getIn(['metrics', 'node']));
+  const metrics = useSelector(({ UI }: RootState) => UI.getIn(['metrics', 'node']));
```
Comment (TheRealJon) [inline frontend/packages/console-app/src/components/nodes/NodesPage.tsx:241]: Wherever we make these type definition updates, we should use the built-in `useSelector` types:
```suggestion
  const metrics = useSelector<RootState, string>(({ UI }) => UI.getIn(['metrics', 'node']));
```

**Review by TheRealJon — COMMENTED** [review]
Just one comment about type definitions for useSelector

File: frontend/packages/console-app/src/components/nodes/NodesPage.tsx (hunk 1 above)
Comment (rhamilto) [reply frontend/packages/console-app/src/components/nodes/NodesPage.tsx]: Thanks, @TheRealJon.  I made a pass at 'em.  Not 100% on all the return types (and even used `any` on a few that I noted with a comment).

**Comment by rhamilto** [comment]
More fixes that cannot be committed as all errors are not yet resolved: [fixes.patch](https://github.com/openshift/console/files/15177731/fixes.patch)

**Pull request closed without merging**